app/src/main/java/com/epmedu/animeal/component/AppSettingsProviderImpl.kt

```diff
@@ -0,0 +1,54 @@
+package com.epmedu.animeal.component
+
+import androidx.datastore.core.DataStore
+import androidx.datastore.preferences.core.MutablePreferences
+import androidx.datastore.preferences.core.Preferences
+import androidx.datastore.preferences.core.booleanPreferencesKey
+import androidx.datastore.preferences.core.edit
+import com.epmedu.animeal.common.component.AppSettings
+import com.epmedu.animeal.common.component.AppSettingsProvider
+import com.epmedu.animeal.common.component.AppSettingsUpdateScope
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.map
+
+private const val INITIAL_GEOLOCATION_PERMISSION = "InitialGeolocationPermissionOnHomeScreen"
```

Comment:
Could be inside the class as booleanPreferencesKey

Reply:
I didn't get what you mean

Reply:
I mean this constant is not needed since it is used only as a key for preferences inside AppSettingsProviderImpl class. I suggest to move the string inside the class as a private field. 
Like this: `val initialGeolocationPermissionKey = booleanPreferencesKey("InitialGeolocationPermissionOnHomeScreen")`